_documentation/04-packaging-drivers.md

```diff
@@ -1,4 +1,295 @@
+**This document contains a complete tutorial as well as information for manufacturers with examples for packaging printer/scanner drivers and releasing them to the Snap Store. If you are looking for information regarding the designing of printer/scanner drivers, kindly refer to <a href="../02-designing-printer-drivers/">Tutorial to Design Printer Drivers</a> and <a href="../03-designing-scanner-drivers/">Tutorial to Design Scanner Drivers</a> respectively.**  
+
+<h2 id="introduction"> Introduction </h2>
+
+After designing the printer/scanner application, you need to distribute these applications, so that the users can directly download and install these applications to support the printers and scanners manufactured by your company. 
+
+One of the primary concerns while packaging the driver is **Distribution-independent packaging**.  The classical printer driver binaries were specifically built for Linux distributions because the distribution provides the layout of CUPS, the libraries, and so on. Also, they have different packaging formats DEB, RPM, and many more.
+
+For each distro, drivers need to be built, packaged, and tested separately. It made the task next to impossible to execute. As a result, most of the hardware manufacturers refrained from making these drivers.
+
+But OpenPrinting has reduced the intensity of your task to a great extent as now you only need one driver for all printers, for all distributions, and distribute it through **snaps**.  
+
+<h2 id="snap"> Snaps </h2>
+
+Snaps are app packages for desktop, cloud, and IoT that are easy to install, secure, cross-platform and dependency-free. 
+
+### Benefits of using snaps 
+
+* A snap is a bundle of an app and its dependencies that works without modification across many different Linux distributions.
+
+* Snaps are more secure. Every package with all its libraries and files in its own sandbox(isolated from other snaps and the host system), fine-grained control for communication between packages.
+
+* Snaps are easily discoverable and installable from the Snap Store, an app store with an audience of millions who can browse and install snaps graphically in the Snap Store or from the command-line.
+
+You are encouraged to dive deep into the official <a href="https://snapcraft.io/docs">Snap Documentation</a> to explore more about it.
+
+<h2 id="package"> Packaging Drivers </h2>
+
+Most Snaps we create are driven using `snapcraft.yaml`. This file describes a snap’s build dependencies and run-time requirements, it integrates remote repositories and extensions and runs custom scripts and hooks for better integration with CI systems.
+
+As a distributor, you would create a `snapcraft.yaml` file and upload the same on Snap Store. When the user runs `snap install`, snapcraft will fetch and open the corresponding `snapcraft.yaml`, interpret it, figure out what to do, and build your application.
```

Comment:
As the developer you will not only upload the snapcraft.yaml, but also auxiliary files. See the cups-snap repository on OpenPrinting. All files of it get uploaded. Same also for the hp-printer-app respository. The Snap Store builds the Snap for several architectures (currently 6 AFAIR) and keep them on their servers for download. If a user runs "snap install ..." the package gets downloaded to the user. Note that the build only happens once with the developer's upload and not with each user's download (imaging what would happen with LibreOffice if the Snap Store would build it on each download request).

Reply:
@tillkamppeter I have corrected the mistake. Kindly review and merge the changes.